gui/default/syncthing/core/syncthingController.js

```diff
@@ -1734,6 +1746,23 @@ angular.module('syncthing.core')
             });
         };
 
+        $scope.showRemoteNeed = function (device) {
+            $scope.remoteNeed = {};
+            $scope.remoteNeedFolders = [];
+            var folderIDCache = $scope.deviceFolders(device);
+            for (var i = 0; i < folderIDCache.length; i++) {
+                var pos = 0
+                if ($scope.completion[device.deviceID][folderIDCache[i]].needCount > 0) {
+                    getRemoteNeed(pos, device.deviceID, folderIDCache[i]);
+                    pos++;
```

Comment:
If anyone with any javascript skills can tell me how to more nicely keep folder order despite the fact that http.get is asynchronous, please do so.

Reply:
```
var promises = [];
for (var i = 0; i < folders.length, i++) {
    promises.push($http.get(url))
}
$q.all(promises).then(function (results) {
}).error(function (results) {});
```
https://docs.angularjs.org/api/ng/service/$q

Reply:
To be honest, perhaps I'd contract the folder sections, and fetch as they get expanded.